### PR TITLE
Splitting the Btag SF to FastSim FullSim components

### PR DESCRIPTION
**Pull request by scarletnorberg**

### CODE REVIEW

**Comment by zhenbinwu** [comment]
Do we really need BTagWeightHeavy_FS and BTagWeightLight_FS?

File: python/modules/BtagSFWeightProducer.py [hunk 1]
```diff
@@ -195,19 +204,22 @@ def analyze(self, event):
...
+                BTagWeightN_FS      *= 1 - btagSF_FS * eff
                 BTagWeightN_up   *= 1 - btagSF_up * eff
                 BTagWeightN_down *= 1 - btagSF_down * eff
                 BTagWeightN_up_FS   *= 1 - btagSF_up_FS * eff
                 BTagWeightN_down_FS *= 1 - btagSF_down_FS * eff
 
                 if abs(flavor) == 5:
                         BTagWeightNHeavy      *= 1 - btagSF * eff
+                        BTagWeightNHeavy_FS      *= 1 - btagSF_FS * eff
                         BTagWeightNHeavy_up   *= 1 - btagSF_up * eff
```
Comment (zhenbinwu) [inline python/modules/BtagSFWeightProducer.py]: where is BTagWeightNHeavy_down?

**Comment by zhenbinwu** [comment]
Since they are variables per event, I am fine with store extra

**Comment by scarletnorberg** [comment]
No we do not it was for completeness and knowledge that it would not take up to much space.

File: python/modules/BtagSFWeightProducer.py (hunk 1 above)
Comment (scarletnorberg) [reply python/modules/BtagSFWeightProducer.py]: Let me double check.

Comment (scarletnorberg) [reply python/modules/BtagSFWeightProducer.py]: This is now fixed.

**Comment by zhenbinwu** [comment]
Looks Ok. Merging. And now I have to resubmit all the fastsim signals

File: python/modules/BtagSFWeightProducer.py [hunk 2]
```diff
@@ -151,7 +157,8 @@ def analyze(self, event):
                 eff = self.h_eff_udsg.GetBinContent(pt_bin, eta_bin);
             
             if self.FastSim:
-                btagSF = jet.btagSF*jet.btagSF_FS
+                btagSF = jet.btagSF
+                btagSF_FS=jet.btagSF_FS
```
Comment (zhenbinwu) [inline python/modules/BtagSFWeightProducer.py]: This is missing for fullsim, while the rest of the code call btagSF_FS even for fullsim. I will fix it

**Pull request merged**